KonomiAI/Chronicle
# feat: added readonly capabilities and made customer ui compatible

### Related Issue(s)



resolves #309



### Quick overview of the change

![image](https://user-images.githubusercontent.com/11084692/197421610-8de24532-9570-414b-99c3-1003ab2fdb4b.png)

- Added new `usePermission` hook to automatically see if user has write permission on the page they are in
- Added new permission context to identify where the user is located
- Disabled fields on customers page if user only has read permissions
- Disabled savebar on pages where user can only read

### Quick testing guide

1. Create a new staff with readonly permissions
2. Go to customers page and see that fields are not updatable
